### PR TITLE
[debugger] Removing some asserts

### PR DESCRIPTION
**Pull request by thaystg**
Removing some asserts and returning err_invalid_argument with an error message when it's possible.

Fixes https://github.com/mono/mono/issues/19651

### CODE REVIEW

File: mono/mini/debugger-agent.c [hunk 1]
```diff
@@ -4589,7 +4589,10 @@ get_object_id_for_debugger_method (MonoClass* async_builder_class)
 	ERROR_DECL (error);
 	GPtrArray *array = mono_class_get_methods_by_name (async_builder_class, "get_ObjectIdForDebugger", 0x24, 1, FALSE, error);
 	mono_error_assert_ok (error);
-	g_assert (array->len == 1);
+	if (array->len != 1) {
```
Comment (vargaz) [inline mono/mini/debugger-agent.c:4592]: Some of these asserts are asserting that some methods/fields are not removed by the linker. If they are transformed into errors, the original problem still remains.

Comment (thaystg) [reply mono/mini/debugger-agent.c:4592]: This is exactly this problem: https://github.com/mono/mono/issues/19651
Another implementation of AsyncTaskMethodBuilder as you can see here:
https://github.com/Cysharp/UniTask/blob/master/Assets/UniRx.Async/CompilerServices/AsyncUniTaskMethodBuilder.cs
And this implementation doesn't implements get_ObjectIdForDebugger. That is the reason why I removed the assert and returned an error. Why it couldn't be an error?

Comment (vargaz) [reply mono/mini/debugger-agent.c:4592]: It could return an error, but without that method, async debugging will not work.

**Comment by thaystg** [comment]
@vargaz , I tested on VsMac and with this change if we cannot find get_ObjectIdForDebugger and we cannot find the property Task, we cannot single step the async method, but we can continue debugging and stopping in other breakpoints, does it make sense for you?

**Pull request merged**